src/db-api.js

```diff
@@ -23,34 +31,53 @@ const unspentAddresses = (db: Pool) => async (): Promise<ResultSet> =>
     rowMode: 'array',
   })
 
+const utxoQuery = `SELECT 
+  TRIM(LEADING '\\x' from tx.hash::text) AS "tx_hash", tx_out.index AS "tx_index",
+  tx_out.address AS "receiver", tx_out.value AS "amount", tx.block::INTEGER as "block_num"
+FROM tx
+INNER JOIN tx_out ON tx.id = tx_out.tx_id
+WHERE NOT EXISTS (SELECT true
+  FROM tx_in
+  WHERE (tx_out.tx_id = tx_in.tx_out_id) AND (tx_out.index = tx_in.tx_out_index)
+) AND (tx_out.address = ANY($1))`
+
 /**
  * Queries UTXO table looking for unspents for given addresses
  *
  * @param {Db Object} db
  * @param {Array<Address>} addresses
  */
 const utxoForAddresses = (db: Pool) => async (addresses: Array<string>) =>
-  db.query('SELECT * FROM "utxos" WHERE receiver = ANY($1)', [addresses])
+  db.query({
+    text: utxoQuery,
+    values: [addresses],
+  })
 
 const utxoSumForAddresses = (db: Pool) => async (addresses: Array<string>) =>
-  db.query('SELECT SUM(amount) FROM "utxos" WHERE receiver = ANY($1)', [
-    addresses,
-  ])
-
-// Cached queries
-const txHistoryQuery = (limit: number) => `
-  SELECT *
-  FROM "txs"
-  LEFT JOIN (SELECT * from "bestblock" LIMIT 1) f ON true
-  WHERE 
-    hash = ANY (
-      SELECT tx_hash 
-      FROM "tx_addresses"
-      where address = ANY ($1)
-    )
-    AND last_update >= $2
-  ORDER BY last_update ASC
-  LIMIT ${limit}
+  db.query(`SELECT SUM(amount) FROM (${utxoQuery}) as utxo_table`, [addresses])
+
+const txHistory = (limit: number) => `
```

Comment:
I would retain the `txHistoryQuery` variable name unless there is some strong reason not to - it's a query, not the history itself